cmake/EthDependencies.cmake

```diff
@@ -25,6 +25,9 @@ set(ETH_SCRIPTS_DIR ${ETH_CMAKE_DIR}/scripts)
 ## use multithreaded boost libraries, with -mt suffix
 set(Boost_USE_MULTITHREADED ON)
 option(Boost_USE_STATIC_LIBS "Link Boost statically" ON)
+if(WIN32)
+	option(Boost_USE_STATIC_RUNTIME "Link Boost against static C++ runtime libraries" ON)
```

Comment:
I'm actually not sure if this should default to ``OFF`` and only be set to ``ON`` in CI...

Reply:
Then again the boost binary distribution also doesn't have static libraries in general, so the same holds for ``Boost_USE_STATIC_LIBS`` above...

The problem is that linking entirely statically will only happen if one additionally uses ``-DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded`` - but that again only works with CMake 3.16+ :-)...

I think otherwise building will also work, but then boost will link against a static C++ runtime while solidity will link against a dynamic one, which is asking for trouble, unless both are compiled together using the exact same libraries...

Not sure how large the actual danger in all this is, though - and we can just take care to document all of this properly... So I guess we can merge this PR regardless, but I thought better to mention this - and we should at least update the docs soon after and before the next release.

Reply:
But defaulting to ``ON`` here will always break the existing AppVeyor builds, if we don't change the config for those :-)...

Reply:
I just pushed another commit setting that for AppVeyor to at least give it a chance :-).